index.ts

```diff
@@ -53,12 +54,30 @@ export class HawkTracerClient {
             this._client = new NativeClient(option);
         }
         else {
-            this._client = new NativeClient(option.source, option.map_files);
+            this._client = new NativeClient(option.source, option.map_files, option.wait_for_server);
         }
     }
 
-    public start(): boolean {
-        return this._client.start();
+    public start(): Promise<boolean> {
+        const tryConnect = (resolve: any, reject: any) => {
+            try {
+                if (this._client.start()) {
+                    resolve(this);
+                }
+                else {
+                    setTimeout(() => {
```

Comment:
This should be cancelled when `stop()` is called.

Reply:
Good point – we should clearTimeout

Reply:
Should be done now